src/bilder/images/forum/deploy.py

```diff
@@ -0,0 +1,167 @@
+import os
+from pathlib import Path
+
+from pyinfra import host
+from pyinfra.operations import files, server
+
+from bilder.components.hashicorp.consul.models import (
+    Consul,
+    ConsulAddresses,
+    ConsulConfig,
+    ConsulService,
+    ConsulServiceTCPCheck,
+)
+from bilder.components.hashicorp.consul.steps import proxy_consul_dns
+from bilder.components.hashicorp.consul_template.models import (
+    ConsulTemplate,
+    ConsulTemplateConfig,
+    ConsulTemplateTemplate,
+    ConsulTemplateVaultConfig,
+)
+from bilder.components.hashicorp.consul_template.steps import (
+    consul_template_permissions,
+)
+from bilder.components.hashicorp.steps import (
+    configure_hashicorp_product,
+    register_services,
+)
+from bilder.components.hashicorp.vault.models import (
+    Vault,
+    VaultAgentCache,
+    VaultAgentConfig,
+    VaultAutoAuthAWS,
+    VaultAutoAuthConfig,
+    VaultAutoAuthFileSink,
+    VaultAutoAuthMethod,
+    VaultAutoAuthSink,
+    VaultConnectionConfig,
+    VaultListener,
+    VaultTCPListener,
+)
+from bilder.facts.has_systemd import HasSystemd
+from bilder.lib.linux_helpers import DOCKER_COMPOSE_DIRECTORY
+from bridge.lib.magic_numbers import VAULT_HTTP_PORT
+from bridge.lib.versions import CONSUL_TEMPLATE_VERSION, CONSUL_VERSION, VAULT_VERSION
+from bridge.secrets.sops import set_env_secrets
+
+set_env_secrets(Path("consul/consul.env"))
+
+TEMPLATES_DIRECTORY = Path(__file__).resolve().parent.joinpath("templates")
+FILES_DIRECTORY = Path(__file__).resolve().parent.joinpath("files")
+VERSIONS = {
+    "consul": os.environ.get("CONSUL_VERSION", CONSUL_VERSION),
+    "vault": os.environ.get("VAULT_VERSION", VAULT_VERSION),
+    "consul-template": os.environ.get(
+        "CONSUL_TEMPLATE_VERSION", CONSUL_TEMPLATE_VERSION
+    ),
+}
+
+
+files.put(
+    name="Place the forum docker-compose.yaml file",
+    src=str(Path(__file__).resolve().parent.joinpath("files", "docker-compose.yaml")),
+    dest=DOCKER_COMPOSE_DIRECTORY,
+    mode="0660",
+)
+
+files.put(
+    name="Place the forum .env file",
+    src=str(Path(__file__).resolve().parent.joinpath("files", "forum.env.j2")),
+    dest="/etc/consul-template/templates.d/",
+    mode="0660",
+)
+DEPLOYMENT = os.environ.get("DEPLOYMENT", "")
```

Comment:
If we want to ensure that this value is set then it might be good to change to using `os.environ["DEPLOYMENT"]` so that if it's not present then it fails loudly with a `KeyError`

Reply:
mitxonline, mitx, xpro? Any other that I am missing?

Reply:
mitx-staging is the other one